src/api/app/helpers/webui/request_helper.rb

```diff
@@ -100,4 +100,14 @@ def target_project_link(row)
     end
     result
   end
+
+  def render_file_state(file_state)
```

Comment:
I don't see the point of having a helper for using it in only one place (also this implementation is tied to a width of 10% inside a table). 

If you want to keep the helper because is great and could be reusable... then I rather prefer to make this independent of the place where it is going to be used. Maybe using a `span` element for having the style instead of the `td` and even better use CSS for it instead of inline styles.

BTW, the actual implementation could be better like this:

```ruby
def render_file_state(file_state)
  colors = { 'Added' => 'green', 'Deleted' => 'red' }
  "<td style=\"width: 10%; color: #{colors[file_state]}"\"> #{file_state}</td>"
end

Reply:
@mdeniz thanks for your tips, I changed it to your implementation with the span w/o the width.

I chose to use a helper function because I did not want to have that logic in the .haml file but that was also because I did not think of using a `<span>`. I don't know if this is going to be used anywhere else, should I try to move it back to the haml file?

Reply:
> should I try to move it back to the haml file?

This is up to you 😄  I mean, having the login in a helper is ok.

The other thing is using CSS classes instead of the span + inline style. 
That way the helper will only calculate which class is the right one, and you can use it at the td in the haml.

Reply:
Something like this,

In the HAML file:
```haml
   %td{ style: 'width: 10%', class: "file_state_#{file_element['state']}" }= file_state
```

And then in the CSS file:
```css
.file_state_added {
  color: green;
}
.file_state_deleted {
  color: red;
}
```